### ISSUE TITLE
Suggestions for features/functionality.

### ISSUE THREAD

**Issue by BlameTheNetwork** [BlameTheNetwork/NetPi#1]
Feel free to post up ideas and thoughts...

**Comment by akhepcat** [comment]
visual traceroute, of course!

**Comment by pcn** [comment]
Continually run some procedures, and trigger logs on particular patterns events (e.g. something changed, something went from empty -> not empty, or not empty -> empty, etc.).  I'd write this in python with the excellent sh module, but the idea would be to have a configuration of event_type and pattern:action [arguments] where e.g. lldpinfo could be watched, and if certain neighbors change, send  before+after snapshots to syslog. and other such things.

**Comment by BlameTheNetwork** [comment]
To make sure I got it right.....

So a loop of LLDP / CDP and any other relevant functions reading and compare their results to the last run (perhaps store results once in a small txt file and compare to it every time, only changing it if the results are different. As well if results are different, the following happens:

if results change {
write a log entry to an overall log file with following format:
[TIMESTAMP] [LLDP|CDP|ETC] CHANGE: WAS X -> CHANGED TO X.
}

**Comment by pcn** [comment]
Yes, with the ability to specify the match so that things that are supposed to change like timestamp don't kill you.

**Comment by Top-Goodman** [comment]
Hello,
I was hoping to build this onto one of my Pi. I saw online this is supposed to work on a Pi3,
Whenever I sent either image down to my SD card (I've used both Etcher and Win32Diskmaker) I just boot to a rainbow screen. I have tried adding boot_delay=1 or boot_delay=30 to my boot.config.txt I am also using a powercable that works for Raspberry Pi (5.2V x 3A) as I have used this before with other projects. Am I missing something. Is there a way to just build NetPi on top of a Rasbian or NOOBS or something else like this.
Thank you for reading this and I look forward to any (hopefully helpful) replies!
